File: src/main/java/rag/trace/ConsoleTraceSink.java
package rag.trace;

public class ConsoleTraceSink implements TraceSink {

    // prints the trace record of the event in the console
    @Override
    public void record(TraceEvent event) {
        String output = String.format(
            "| %s | %dms | %s | %s",
            event.stage,
            event.durationMs,
            event.summary,
            event.error != null ? "[HATA: " + event.error + "]" : ""
        );
        System.out.println(output);
    }
}

File: src/main/python/rag/trace/console_trace_sink.py
from rag.trace.trace_sink import TraceSink
from rag.trace.trace_event import TraceEvent


class ConsoleTraceSink(TraceSink):
    def accept(self, event: TraceEvent) -> None:
        error_str: str = event.error if event.error else "-"
        print(f"[TRACE] {event.stage} | summary={event.summary} | duration={event.duration_ms}ms | error={error_str}")






